core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -544,6 +545,8 @@ final SearchContext createContext(ShardSearchRequest request, @Nullable Engine.S
                 indexService,
                 indexShard, scriptService, pageCacheRecycler, bigArrays, threadPool.estimatedTimeInMillisCounter(), parseFieldMatcher,
                 defaultSearchTimeout, fetchPhase);
+        context.getQueryShardContext().setFieldStatsProvider(new FieldStatsProvider(engineSearcher, indexService.mapperService()));
```

Comment:
I wonder if we should only do this if the cache is enabled. for all other operations its unneeded and might be overhead? - I mean not the rewrite but the setting the field stats provider.  We check this with `IndicesService #canCache(ShardSearchRequest request, SearchContext context)`